shotover-proxy/src/message/mod.rs

```diff
@@ -173,6 +173,47 @@ impl Message {
         }
     }
 
+    pub fn as_encodable(&self, expected_message_type: MessageType) -> Result<EncodableRef> {
```

Comment:
Now that I've taken a closer a look at how we use this is in CassandraSinkSingle this suggestion feels like overkill. Its possible the extra functionality here might be useful in the future but I would rather stick with a simpler API and refactor it later if needed. Theres also the problem of confusing the meaning of Encodable, as its now actually used for decoding!

So I'm thinking we would be better off with something like:
```rust
/// Placeholder comment: only use straight out of the oven
pub fn as_raw_bytes(self) -> Option<&Bytes> {
    match &self.inner {
        MessageInner::RawBytes { bytes, .. } => Some(bytes),
        _ => None,
    }
}
```

A reason I dont want to expose the inner field as public API is because we might want to add hidden caching to Message in the future and we cant do both of these.

My assumption here is that MessageInner::RawBytes is the only case we will need this for as its the only time we could possibly make an assumption about the state of a message.
In any other case we would be better off just using `.frame()`.